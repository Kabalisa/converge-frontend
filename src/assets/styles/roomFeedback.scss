@import './variables';

.room-feedback {
  font-family: $font-family;
  width: 90%;
  margin: 0 auto;

  &__control {
    height: 70px;
    text-align: right;
    margin-top: 4%;
    margin-bottom: 4%;
  }

  button {
    border: 1px solid map-get($color-palette, primary);
    border-radius: 2px;
    font-size: 16px;
    color: map-get($color-palette, primary);
    padding: 1% 3%;
    margin-left: 0;
    background-color:map-get($color-palette, white);

    @media screen and (max-width: 425px) {
      font-size: 1.05em;
    }
  }

  &__list {
    table {
      clear: left;
      table-layout: fixed;
      width: 100%;
      border-collapse: collapse;
      text-align: left;

      .first-col {
        width: 50%;

        @media screen and (max-width: 425px) {
          width: 24%;
        }

        @media screen and (min-width: 500px) and (max-width: 1023px) {
          width: 35%;
        }
      }

      .last-col {
        @media screen and (max-width: 425px) {
          width: 30%;
        }

        @media screen and (min-width: 500px) and (max-width: 1023px) {
          width: 33%;
        }
      }

      tr {
        padding: 1%;
        border-bottom: 10px solid $background;
        font-weight: 50;
        display: table-row !important;
      }

      th {
        color: map-get($black, 100);
        padding-left: 2%;
        font-weight: inherit;
      }

      th:nth-of-type(2) {
        padding-left: 2.6%;
      }

      td {
        background: map-get($color-palette, white);
        padding: 2%;
        color: map-get($black, 600);
        font-size: 95%;
        vertical-align: bottom;
        text-align: center;
        text-transform: capitalize;

        & img:nth-child(2) {
          padding-left: 20px;
        }

        & img {
          margin-top: 2px;
        }
      }

      td:nth-of-type(1) {
        text-align: left;
        color: map-get($color-palette, primary)
      }

      td:nth-of-type(4), td:nth-of-type(5) {
        padding: 2% 0;
      }

      @media screen and (max-width: 425px) {
        width: 100%;
        margin-right: 0;
      }

      @media screen and (min-width: 768) and (max-width: 1023px) {
        width: 100%;
      }
    }
  }
}
