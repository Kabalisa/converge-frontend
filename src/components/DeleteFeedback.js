import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MrmModal from './commons/Modal';
import '../assets/styles/deleteModal.scss';

/**
 * Delete Feedback Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
class DeleteFeedback extends Component {
  state = {
    closeModal: false,
  };

  /**
   * It closes a modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * Handles the state changes for the deleting feedback modal
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * Handles deleting feedback
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleDeleteFeedback = (event) => {
    event.preventDefault();
    // submit logic goes here
    this.handleCloseModal();
  };

  render() {
    const { closeModal } = this.state;
    return (
      <MrmModal
        className="delete-modal"
        title="Delete Question"
        buttonText="Delete"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
      >
        <div className="delete-modal-content">
          <p id="confirm-msg">
            Are you sure you want to delete {`"${this.props.question}"`}?
            <br />
            This cannot be undone
          </p>
          <div className="modal-actions">
            <button id="cancel-btn" onClick={this.handleCloseModal}>
              Cancel
            </button>
            <button id="delete-btn" onClick={this.handleDeleteFeedback}>
              Delete
            </button>
          </div>
        </div>
      </MrmModal>
    );
  }
}

DeleteFeedback.propTypes = {
  question: PropTypes.string.isRequired,
};

export default DeleteFeedback;
