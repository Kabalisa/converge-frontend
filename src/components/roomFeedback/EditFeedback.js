import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Input } from '../commons/Input';
import Select from '../commons/SelectInput';
import MrmModal from '../commons/Modal';
import ActionButtons from '../commons/ActionButtons';
import '../../assets/styles/editfeedback.scss';

/**
 * Edit Feedback Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
class EditFeedback extends Component {
  state = {
    question: this.props.question,
    type: this.props.type,
    startDate: this.props.startDate,
    duration: this.props.duration,
    closeModal: false,
  };

  /**
   * It closes a modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * It changes the state of the modal
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * Ensures that the state is updated basing
   * on the changes in the input fields
   *
   * @param {Object} target
   *
   * @returns {void}
   */
  handleInputChange = ({ target: { name, value } }) => {
    this.setState({ [name]: value });
  };

  /**
   * Handles editing feedback
   *
   * @param {object} event
   *
   * @returns {void}
   */
  handleEditFeedback = (event) => {
    event.preventDefault();
    /** submit logic here */
    this.handleCloseModal();
  };

  render() {
    const {
      question, type, startDate, duration, closeModal,
    } = this.state;

    return (
      <MrmModal
        title="Edit Question"
        buttonText="Edit"
        closeModal={closeModal}
        handleCloseRequest={this.handleModalStateChange}
        className="edit-feedback-modal"
        modalButtonClassName="edit-button"
      >
        <form className="modal-form" onSubmit={this.handleEditFeedback}>
          <Input
            labelName="Question"
            labelClass="label1"
            inputClass="input1"
            name="question"
            id="question"
            value={question}
            onChange={this.handleInputChange}
          />
          <Select
            labelText="Type"
            name="type"
            id="type"
            value={type}
            onChange={this.handleInputChange}
            options={[]}
            wrapperClassName="label1"
            placeholder="Rate"
            selectInputClassName="edit-feedback-select"
          />
          <Input
            labelName="Start Date"
            labelClass="label1"
            inputClass="input1"
            name="startDate"
            id="startDate"
            value={startDate}
            placeholder="03 Nov 2018"
            onChange={this.handleInputChange}
          />
          <Input
            labelName="Duration"
            labelClass="label1"
            inputClass="input1"
            name="duration"
            id="duration"
            value={duration}
            placeholder="3 Weeks"
            onChange={this.handleInputChange}
          />
          <ActionButtons
            withCancel
            onClickCancel={this.handleCloseModal}
            actionButtonText="Save Changes"
          />
        </form>
      </MrmModal>
    );
  }
}

EditFeedback.propTypes = {
  question: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  duration: PropTypes.string.isRequired,
};

export default EditFeedback;
