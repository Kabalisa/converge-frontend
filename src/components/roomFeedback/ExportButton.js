import React, { Fragment } from 'react';
import '../../../src/assets/styles/exportButton.scss';

/**
 * Represents an export button
 * @returns {JSX}
 * @constructor
 */
const ExportButton = () => (
  <Fragment>
    <span className="export-button">
      <svg width="10px" height="10px" viewBox="0 0 433.5 433.5" className="svg-export">
        <path d="M395.25 153h-102V0h-153v153h-102l178.5 178.5L395.25 153zm-357 229.5v51h357v-51h-357z" />
      </svg>
    Export
    </span>
  </Fragment>
);

export default ExportButton;
