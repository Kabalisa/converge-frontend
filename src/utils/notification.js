/* eslint-disable */
/**
 * toastr => This is the toastr module imported from 'toastr' npm package
 * notifyType => This is the notification type (success or error)
 * message => This is the notification message
 *
 * How to use this function
 * const notify = notification(toastr, 'success', 'noficationMessage');
 * notify()
 */
const notification = (toastr, notifyType, message) => {
  toastr['options'] = {
    'closeButton': true,
    'positionClass': 'toast-top-center',
    'preventDuplicates': true,
    'onclick': null,
    'showDuration': '300',
    'hideDuration': '1000',
    'timeOut': '5000',
    'extendedTimeOut': '100000',
    'showEasing': 'swing',
    'hideEasing': 'linear',
    'showMethod': 'fadeIn',
    'hideMethod': 'fadeOut',
  };
  return () => toastr[notifyType](message);
};

export default notification;