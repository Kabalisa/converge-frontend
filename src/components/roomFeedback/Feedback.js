/* eslint react/no-array-index-key: 0 */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import CheckBoxSlide from '../commons/CheckboxSlide';
import EditFeedbackComponent from './EditFeedback';
import { UPDATE_QUESTION_STATUS_MUTATION } from '../../graphql/mutations/Question';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../../src/graphql/queries/questions';
import DeleteFeedback from './DeleteFeedback'; // eslint-disable-line

/**
 * Feedback Component
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const Feedback = (props) => {
  const { role } = props;
  return (
    props.feedback.map(({
      id, question, startDate, endDate, questionTitle,
      questionResponseCount, questionType, isActive,
    }) => (
      <div className="table__row" key={id}>
        <span>
          {question}
        </span>
        <span>{questionType}</span>
        <span>{questionResponseCount}</span>
        <span>{props.startDateFormatter(startDate)}</span>
        <span>{props.durationFormatter(startDate, endDate)}</span>
        {
          role === '2' &&
          <Fragment>
            <span>
              {!isActive &&
                <EditFeedbackComponent
                  id="edit-modal"
                  questionId={id}
                  question={question}
                  questionType={questionType}
                  questionTitle={questionTitle}
                  startDate={startDate}
                  endDate={endDate}
                />}
              <DeleteFeedback id={id} question={question} />
            </span>
            <span className="checkbox">
              <CheckBoxSlide
                questionId={parseInt(id, 10)}
                isChecked={isActive}
                actionToPerform="mutation"
                mutationQuery={UPDATE_QUESTION_STATUS_MUTATION}
                updateQuery={GET_ROOM_FEEDBACK_QUESTIONS_QUERY}
              />
            </span>
          </Fragment>
        }
      </div>
    )));
};

Feedback.defaultProps = {
  feedback: [
    {
      question: 'There is no question so far',
      questionType: 'Input',
      startDate: '2019-02-21 23:42:43',
      endDate: '2019-02-21 23:42:43',
      questionResponseCount: 0,
      isActive: false,
    },
  ],
};

Feedback.propTypes = {
  feedback: PropTypes.arrayOf(PropTypes.object).isRequired,
  role: PropTypes.string.isRequired,
};

export default Feedback;
