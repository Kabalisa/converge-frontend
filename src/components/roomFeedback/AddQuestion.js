/* eslint react/no-array-index-key: 0 */
import React, { Component } from 'react';
import moment from 'moment';
import { graphql } from 'react-apollo';
import toastr from 'toastr';
import PropTypes from 'prop-types';
import notification from '../../utils/notification';
import Calendar from '../commons/Calendar';
import MrmModal from '../commons/MrmModal';
import { SelectInput as Select } from '../commons';
import '../../assets/styles/addQuestion.scss';
import { ADD_ROOM_FEEDBACK_QUESTIONS } from '../../graphql/mutations/Question';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../../src/graphql/queries/questions';


/**
 * Add Question Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
export class AddQuestion extends Component {
  state = {
    question: '',
    questionTitle: '',
    questionType: 0,
    startDate: moment().format('MMM DD Y'),
    endDate: moment().format('MMM DD Y'),
    startTime: moment().format('HH:MM'),
    endTime: moment()
      .add(1, 'hours')
      .format('HH:MM'),
    calendarOpen: false,
    closeModal: false,
    isLoading: false,
    error: {},
    options: [
      {
        id: '1',
        name: 'Rate',
      },
      {
        id: '2',
        name: 'Check',
      },
      {
        id: '3',
        name: 'Suggest',
      },
    ],
  };

  /**
   * 1. change isLoading state to it's opposite value
   * i.e true to false or vise verser
   *
   * @returns {void}
   */
  toggleLoading = () => {
    this.setState({
      isLoading: !this.state.isLoading,
    });
  }

  /**
   * It toggles the visibility of the modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * It toggles the visibility of the calendar modal
   *
   * @returns {void}
   */
  calenderToggle = () => {
    const { calendarOpen } = this.state;
    this.setState({ calendarOpen: !calendarOpen });
  };

  /**
   * It updates the state with the selected start and end date
   *
   * @returns {void}
   */
  sendDateData = (start, end) => {
    this.setState({ startDate: start, endDate: end });
    this.calenderToggle();
  };

  /**
   * It updates the state value of closeModal
   * to false whenever the modal closes
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  }

  /**
   * It handles the selected choice from the options
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  handleInputChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  /**
   * This function formats the returned date
   * and time to form a datetime object for
   * both start date and end date
   *
   * @param {date} theDate
   * @param {time} theTime
   *
   * @returns {string}
   */
  formatDateTime = (theDate, theTime) => {
    const theDateValue = (`0${new Date(theDate).getDate()}`).slice(-2);
    const theMonth = (`0${parseInt(new Date(theDate).getMonth() + 1, 10)}`).slice(-2);
    const theYear = new Date(theDate).getFullYear();
    const finalDate = `${theYear}-${theMonth}-${theDateValue}T${theTime}:00`;
    return finalDate;
  };

  /**
   * handles creation of a question
   *
   * @returns {void}
   */
  createQuestion = () => {
    const {
      question, questionTitle, questionType, startDate, endDate, startTime, endTime, options,
    } = this.state;
    const valueOfQuestionType = parseInt(questionType, 10) - 1;
    const questionTypeName = options[valueOfQuestionType].name;
    this.toggleLoading();
    this.props.addQuestion({
      variables: {
        question,
        questionTitle,
        questionType: questionTypeName,
        startDate: this.formatDateTime(startDate, startTime),
        endDate: this.formatDateTime(endDate, endTime),
      },
    })
      .then(() => {
        notification(
          toastr,
          'success',
          'Question has been added successfully',
        )();
        this.setState({
          question: '',
          questionTitle: '',
          questionType: 0,
          startDate: moment().format('MMM DD Y'),
          endDate: moment().format('MMM DD Y'),
          startTime: moment().format('HH:MM'),
          endTime: moment().format('HH:MM'),
        });
        this.handleCloseModal();
        this.toggleLoading();
      })
      .catch((err) => {
        this.toggleLoading();
        this.handleCloseModal();
        notification(toastr, 'error', err.graphQLErrors[0].message)();
      });
  }

  /**
   * compares the start date with today's date
   * compares the end date with today's date
   * compares the start date with end date
   *
   * @param {date} start
   * @param {date} end
   *
   * @returns {string}
   */
  compareDates = (start, end) => {
    const today = new Date();
    today.setHours(0, 0, 0, 0);
    if (start < today) {
      return this.inputErrors('date', 'Start date cannot be past days');
    } else if (end < today) {
      return this.inputErrors('date', 'End date should be at least a day after today');
    } else if (end <= start) {
      return this.inputErrors('date', 'End date should be at least a day after start date');
    }
    return this.createQuestion();
  }

  /**
   * validates question input before creating a question
   *
   * @param {Object} event
   *
   * @returns {void}
   */
  validateInputFields = () => {
    const {
      question, questionTitle, questionType, startDate, endDate,
    } = this.state;
    if (!questionTitle) {
      return this.inputErrors('questionTitle', 'Please provide the title of the question');
    } else if (!question) {
      return this.inputErrors('question', 'Please provide the feedback question');
    } else if (questionType === 0) {
      return this.inputErrors('type', 'Please select the question type');
    }
    return this.compareDates(new Date(startDate), new Date(endDate));
  }

  /**
   * renders the validation errors for input fields
   *
   * @param {string} errorKey
   * @param {string} message
   *
   * @returns {void}
   */
  inputErrors = (errorKey, message) => {
    this.setState({
      ...this.state.error[errorKey] = message,
    });
  }

  /**
   * It returns the inputs for adding start time and endtime of a question
   * @param {Function} handleInputChange
   *
   * @returns {JSX}
   */
  renderTimeInputs = () => (
    <div className="inputs-inline">
      <div className="inputs">
        <label htmlFor="startTime" className="inputs-title">
          <span>Start</span>
        </label>
        <input
          type="time"
          name="startTime"
          onChange={this.handleInputChange}
          id="timeField"
          value={this.state.startTime}
        />
      </div>
      <div className="inputs">
        <label htmlFor="endTime" className="inputs-title">
          <span>End</span>
        </label>
        <input
          type="time"
          name="endTime"
          onChange={this.handleInputChange}
          id="timeField"
          value={this.state.endTime}
        />
      </div>
    </div>
  );

  /**
   * It returns the input text box for typing the question title
   *
   * @returns {JSX}
   */
  renderQuestionTitleInputBox = () => (
    <div>
      {this.state.error.questionTitle &&
      <span className="span-error">
        <br />{this.state.error.questionTitle}
      </span>}
      <div className={this.state.error.questionTitle ? 'input-error' : 'inputs'}>
        <input
          placeholder="Enter Question Title Here..."
          type="text"
          name="questionTitle"
          onChange={this.handleInputChange}
          value={this.state.questionTitle}
        />
      </div>
    </div>
  );

  /**
   * It returns the input text box for typing the question
   *
   * @returns {JSX}
   */
  renderQuestionInputBox = () => (
    <div>
      {this.state.error.question &&
        <span className="span-error">
          <br />{this.state.error.question}
        </span>}
      <div className={this.state.error.question ? 'input-error' : 'inputs'}>
        <input
          placeholder="Enter Question here..."
          type="text"
          name="question"
          onChange={this.handleInputChange}
          value={this.state.question}
        />
      </div>
    </div>
  );

  /**
   * It renders the calendar component
   *
   * @returns {JSX}
   */
  renderCalendarComponent = () => (
    <div>
      {this.state.error.date &&
        <span className="span-error">
          <br />{this.state.error.date}
        </span>}
      <Calendar
        sendData={this.sendDateData}
        classProp={this.state.error.date ? 'calendarIconBtn-error' : 'calendarIconBtn'}
      />
    </div>
  );


  /**
   * It renders the dropdown for
   * selecting question type
   *
   * @returns {JSX}
   */
  rendersQuestionType = () => (
    <div>
      {this.state.error.type &&
      <span className="span-error">
        <br />{this.state.error.type}
      </span>}
      <Select
        labelText=""
        name="questionType"
        id={this.state.error.type ? 'selectType-error' : 'selectType'}
        value={this.state.questionType}
        onChange={this.handleInputChange}
        wrapperClassName="input-wrapper"
        placeholder="Select Type"
        options={this.state.options}
      />
    </div>
  );

  /**
   * It shows all the input fields required for creating a question
   *
   * @returns {JSX}
   */
  renderQuestionValues = () => (
    <div className="question-form">
      <span className="question-form__sections">Date</span>
      {this.renderCalendarComponent()}
      <span className="question-form__sections">Time</span>
      {this.renderTimeInputs()}
      <span className="question-form__sections">Question Title</span>
      {this.renderQuestionTitleInputBox()}
      <span className="question-form__sections">Question</span>
      {this.renderQuestionInputBox()}
      <span className="question-form__sections">Question Type</span>
      {this.rendersQuestionType()}
    </div>
  );

  render() {
    return (
      <MrmModal
        title="ADD QUESTION"
        buttonText="Add Question"
        modalButtonClassName="add-question-button"
        modalContent={this.renderQuestionValues()}
        cancelButtonText="CANCEL"
        actionButtonText="ADD QUESTION"
        handleSubmit={this.validateInputFields}
        isLoading={this.state.isLoading}
      />

    );
  }
}
AddQuestion.defaultProps = {
  addQuestion: null,
};
AddQuestion.propTypes = {
  addQuestion: PropTypes.func,
};
export default graphql(ADD_ROOM_FEEDBACK_QUESTIONS,
  {
    name: 'addQuestion',
    options:
  { refetchQueries: [{ query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY }] },
  })(AddQuestion);
