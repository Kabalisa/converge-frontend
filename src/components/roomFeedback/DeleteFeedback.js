import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import toastr from 'toastr';
import Modal from '../commons/MrmModal';
import '../../assets/styles/deleteModal.scss';
import notification from '../../utils/notification';
import { DELETE_ROOM_FEEDBACK_QUESTION } from '../../graphql/mutations/Question';
import GET_ROOM_FEEDBACK_QUESTIONS_QUERY from '../../graphql/queries/questions';

/**
 * Delete Feedback Component
 *
 * @extends React.Component
 *
 * @returns {JSX}
 */
export class DeleteFeedback extends Component {
  state = {
    closeModal: false,
    isDeleting: false,
  };

  /**
   * It closes a modal
   *
   * @returns {void}
   */
  handleCloseModal = () => {
    this.setState({ closeModal: true });
  };

  /**
   * Handles the state changes for the deleting feedback modal
   *
   * @returns {void}
   */
  handleModalStateChange = () => {
    this.state.closeModal && this.setState({ closeModal: false });
  };

  /**
   * 1. change isLoading state to it's opposite value
   * i.e true to false or vise verser
   *
   * @returns {void}
   */
  toggleLoading = () => {
    this.setState({
      isDeleting: !this.state.isDeleting,
    });
  }

  /**
   * Handles deleting feedback
   *
   * @returns {void}
   */
  handleDeleteFeedback = () => {
    const { id, question, deleteFeedback } = this.props;
    this.toggleLoading();
    deleteFeedback({
      variables: {
        questionId: id,
      },
      refetchQueries: [{ query: GET_ROOM_FEEDBACK_QUESTIONS_QUERY }],
    })
      .then(() => {
        this.toggleLoading();
        this.handleCloseModal();
        notification(toastr, 'success', `${question} has been deleted successfully`)();
      })
      .catch((err) => {
        this.toggleLoading();
        this.handleCloseModal();
        notification(toastr, 'error', err.graphQLErrors[0].message)();
      });
  };

  /**
   * renders model content for delete feedback
   *
   * @returns {JSX}
   */

   renderModalContent = () => (
     <div className="delete-modal-content">
       <p id="confirm-msg">
         Are you sure you want to delete {`"${this.props.question}"`} ?
         <br />
         This cannot be undone
       </p>
     </div>
   )

   render() {
     const { isDeleting } = this.state;

     return (
       <Modal
         title="DELETE QUESTION"
         buttonText="Delete"
         actionButtonText="DELETE QUESTION"
         modalContent={this.renderModalContent()}
         isLoading={isDeleting}
         handleSubmit={this.handleDeleteFeedback}
         cancelButtonText="CANCEL"
       />


     );
   }
}

DeleteFeedback.propTypes = {
  question: PropTypes.string.isRequired,
  deleteFeedback: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

export default compose(
  graphql(DELETE_ROOM_FEEDBACK_QUESTION, { name: 'deleteFeedback' }),
  graphql(GET_ROOM_FEEDBACK_QUESTIONS_QUERY, { name: 'allQuestions' }),
)(DeleteFeedback);
