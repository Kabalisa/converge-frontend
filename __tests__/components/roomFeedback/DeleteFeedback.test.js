import React from 'react';
import { shallow } from 'enzyme';
import { DeleteFeedback } from '../../../src/components/roomFeedback/DeleteFeedback';
import notification from '../../../src/utils/notification';

jest.mock('../../../src/utils/notification');

describe('DeleteFeedback Component', () => {
  const initProps = {
    id: '1',
    question: 'Test question',
    deleteFeedback: jest.fn(),
  };
  const wrapper = shallow(<DeleteFeedback {...initProps} />);

  it('should handleCloseModal()', () => {
    wrapper.instance().handleCloseModal();
    expect(wrapper.state('closeModal')).toEqual(true);
  });

  it('should handleModalStateChange()', () => {
    wrapper.instance().handleModalStateChange();
    expect(wrapper.state('closeModal')).toBe(false);
  });

  it('should call notification with a success message', () => {
    wrapper.setProps({
      deleteFeedback: jest.fn(() => Promise.resolve(notification('success'))),
    });
    wrapper.instance().handleDeleteFeedback();
    expect(notification.mock.calls[0][0]).toBe('success');
  });

  it('should call notification with an error message', () => {
    jest.clearAllMocks();
    wrapper.setProps({
      deleteFeedback: jest.fn(() => Promise.reject(notification('error'))),
    });
    wrapper.instance().handleDeleteFeedback();
    expect(notification.mock.calls[0][0]).toBe('error');
  });
});
