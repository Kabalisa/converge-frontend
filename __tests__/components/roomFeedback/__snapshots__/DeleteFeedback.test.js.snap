// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFeedback Component renders correctly 1`] = `
<MrmModal
  buttonText="Delete"
  className="delete-modal"
  closeModal={false}
  handleCloseRequest={[Function]}
  modalButtonClassName="button"
  title="Delete Question"
>
  <div
    className="delete-modal-content"
  >
    <p
      id="confirm-msg"
    >
      Are you sure you want to delete 
      "How would you rate the cleanliness of this room"
      ?
      <br />
      This cannot be undone
    </p>
    <ActionButtons
      actionButtonText="DELETE QUESTION"
      cancelButtonText="CANCEL"
      isLoading={false}
      onClickCancel={[Function]}
      onClickSubmit={[Function]}
      withCancel={true}
    />
  </div>
</MrmModal>
`;
